server/responders/query.py

```
@@ -75,11 +74,13 @@ def handler(start_response, requestData):
             whereClause.ParameterPlaceHolder = '%s'
             whereClause.Decode(query, True)
             whereClause.CreateSelectStatement()
-
             sqlQuery = "SELECT "
             if distinct:
                 sqlQuery += " DISTINCT "
             sqlQuery += "{0} FROM {1}".format(','.join(columns), DBTBESC(tableId))
+            for join in joins:
+                if 'type' in join and join['type'] in ['', 'INNER', 'LEFT', 'RIGHT', 'FULL']:
```

This is a silent failure if the join is missing or not a known type?